operator/identity/gc_test.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/cilium/cilium/pkg/option"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(
+		m,
+		// To ignore goroutine started from sigs.k8s.io/controller-runtime/pkg/log.go
+		// init function
+		goleak.IgnoreTopFunction("time.Sleep"),
+	)
+}
+
+func TestIdentitiesGC(t *testing.T) {
+	var clientset *k8sClient.FakeClientset
+
+	hive := hive.New(
+		// provide a fake clientset
+		cell.Provide(func() k8sClient.Clientset {
+			clientset, _ = k8sClient.NewFakeClientset()
+			return clientset
+		}),
+
+		// provide resources
+		k8s.SharedResourcesCell,
+
+		// provide identities gc test configuration
+		cell.Provide(func() IdentitiesGCConfig {
+			return IdentitiesGCConfig{
+				IdentityGCInterval:       50 * time.Millisecond,
+				IdentityHeartbeatTimeout: 50 * time.Millisecond,
+
+				IdentityGCRateInterval: time.Minute,
+				IdentityGCRateLimit:    2500,
+			}
+		}),
+		cell.Provide(func() IdentityGCSharedConfig {
+			return IdentityGCSharedConfig{
+				IdentityAllocationMode: option.IdentityAllocationModeCRD,
+				EnableMetrics:          false,
+				ClusterName:            defaults.ClusterName,
+				K8sNamespace:           "",
+				ClusterID:              0,
+			}
+		}),
+
+		// provide the identities gc to be tested
+		cell.Provide(newIdentitiesGC),
+
+		// initial setup for the test
+		cell.Invoke(func(clientset k8sClient.Clientset) {
+			nodes := []*corev1.Node{
+				{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Node",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "identity-control-plane",
+						Labels: map[string]string{"kubernetes.io/hostname": "identity-control-plane"},
+					},
+				},
+				{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Node",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "identity-worker",
+						Labels: map[string]string{"kubernetes.io/hostname": "identity-worker"},
+					},
+				},
+			}
+			for _, node := range nodes {
+				if _, err := clientset.CoreV1().Nodes().
+					Create(context.Background(), node, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("Failed to create node %v: %s", node, err)
+				}
+			}
+
+			identity := &v2.CiliumIdentity{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "cilium.io/v2",
+					Kind:       "CiliumIdentity",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "99999",
+					Labels: map[string]string{
+						"io.cilium.k8s.policy.cluster":        "default",
+						"io.cilium.k8s.policy.serviceaccount": "default",
+						"io.kubernetes.pod.namespace":         "default",
+						"test":                                "identities-gc",
+					},
+				},
+			}
+			if _, err := clientset.CiliumV2().CiliumIdentities().
+				Create(context.Background(), identity, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Failed to create identity %v: %s", identity, err)
+			}
+		}),
+
+		cell.Invoke(func(lc hive.Lifecycle, clientset k8sClient.Clientset, gc *identitiesGC) {
```

Comment:
`clientset` and `gc` don't seem to be referenced here. Could also move these into the test setup function.